core/model/src/main/java/org/eclipse/rdf4j/model/util/Literals.java

```diff
@@ -51,6 +48,11 @@ public static String getLabel(Value v, String fallback) {
 		return v instanceof Literal ? getLabel((Literal) v, fallback) : fallback;
 	}
 
+	public static String getLabel(Optional v, String fallback) {
+
+		return v instanceof Optional ? getLabel((Value) v.orElseGet(null), fallback) : fallback;
```

Comment:
You don't need to check if `v` is an `Optional` - it always is (because that's the type of the argument) - though perhaps you could check that it is not `null`.

Reply:
Yes I fixed that in my latest commit.